### ISSUE TITLE
Improving Annual Parallax Calculation

### ISSUE THREAD

**Issue by ShotaMiyazaki94** [ShotaMiyazaki94/microjax#10]
In the current implementation of the annual microlensing parallax model, the computed source-lens separation agrees with other publicly available packages near the event time $t_0$. However, at epochs significantly offset from $t_0$, discrepancies on the order of $10^{-3}$ in normalized separation units ($\Delta u \sim 10^{-3}$) are observed. While such deviations are negligible under typical ground-based photometric precision, they may become non-negligible for space-based observatories, especially in events exhibiting planetary caustic crossings far from the primary peak magnification. For accurate modeling and interpretation of such high-precision datasets, a refined version of the parallax formalism with improved numerical accuracy should be implemented in a future release.

![Image](https://github.com/user-attachments/assets/568f2e21-3d0a-4958-969b-60c8ef58ec35)

![Image](https://github.com/user-attachments/assets/bb5ba6e0-6114-40f5-83b2-e703fb11d75b)

**Comment by ShotaMiyazaki94** [comment]
I implemented a new JAX-friendly annual-parallax path that reads the bundled JPL Horizons Earth ephemeris (same tables used by `MulensModel` / `VBMicrolensing` / `jacsanomaly`). The new API lives in `microjax.trajectory` as `set_parallax_ephem` + `compute_parallax_ephem`, so we get HJD correction, JIT, and autodiff support.

The test code [parallax.py](https://github.com/user-attachments/files/25017460/parallax.py) now: 
- sets a fiducial event (coords 17:45:40 -29:00:28, u0=0.1, tE=exp(ln 100), piEN=piEE=0.5, t0=7000, ±3 yr, 1000 samples);
- compares four implementations: VBMicrolensing (parallaxsystem=1), MulensModel (Earth-orbital parallax), microjax Keplerian approximation, and the new ephemeris-based microjax path;
- saves parallax_mag_diff.png, parallax_traj_diff.png, and parallax_traj_compare.png in the test directory.

Results: the ephemeris-based microjax track overlays MulensModel to within the plotted residual floor while remaining fully differentiable/JAX-compatible.

<img width="2400" height="800" alt="Image" src="https://github.com/user-attachments/assets/8bd15b92-0e85-4a0d-affa-d2ba46bce52f" />
<img width="1200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/2bb0c795-e568-4bd6-b454-c23f0cf8e4f7" />
<img width="1200" height="800" alt="Image" src="https://github.com/user-attachments/assets/c21fbd1c-f0f2-46f1-85b9-80d0bbde1686" />
